code/controllers/subsystem/daynightcycle.dm

```diff
@@ -0,0 +1,77 @@
+var/datum/subsystem/daynightcycle/SSDayNight
+
+#define TOD_MORNING 	1
+#define TOD_SUNRISE 	2
+#define TOD_DAYTIME 	3
+#define TOD_AFTERNOON 	4
+#define TOD_SUNSET 		5
+#define TOD_NIGHTTIME 	6
+
+/datum/subsystem/daynightcycle
+	name          = "Day Night Cycle"
+	init_order    = SS_INIT_DAYNIGHT
+	display_order = SS_DISPLAY_DAYNIGHT
+	priority      = SS_PRIORITY_DAYNIGHT
+	wait          = 20 MINUTES
+	flags         = SS_BACKGROUND|SS_FIRE_IN_LOBBY
+
+	var/current_color = "#4d3d66"
+	var/timeOfDay = TOD_MORNING
+	var/asspower = 10
+	var/assrange = 1
+
+/datum/subsystem/daynightcycle/New()
+	NEW_SS_GLOBAL(SSDayNight)
+
+/datum/subsystem/daynightcycle/Initialize()
+	var/list/timestwopick = list(TOD_MORNING,
+								TOD_SUNRISE,
+								TOD_DAYTIME,
+								TOD_AFTERNOON,
+								TOD_SUNSET,
+								TOD_NIGHTTIME)
+	timeOfDay = pick(timestwopick)
+	..()
+
+/datum/subsystem/daynightcycle/fire(resumed = FALSE)
+	if(flags & SS_NO_FIRE)
+		return
+	if(!map.daynight_cycle)
+		flags |= SS_NO_FIRE
+		pause()
+	else
+		switch(timeOfDay)
```

Comment:
I notice that all of your times of day are completely lit up. Even nighttime is just purple out. This seems the ideal use of adjusting light power - and turning it off completely for night.

Reply:
Are you sure? it will destroy it in update_light() instead of change it.

Reply:
It's undeniably harder on resources, but it feels a little weird to have night that isn't actually obstructing your view at all!

Reply:
Hey, you actually can't see a lot if the light is dark enough unironically, but I redid it so the lights are non-existent 50% of the time.